### PR TITLE
DEV-2805: Fix payment selection highlight

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a

#### What's this PR do?
Show the correct contribution amount selection if user inputs a custom amount and changes frequency.

#### Why are we doing this? How does it help us?
The user was already contributing with the underlying amount selection, but it wasn't being highlighted correctly in the amount selection section.

#### How should this be manually tested? Please include detailed step-by-step instructions.
1. Enter a contribution page such as [RevEngine](https://alfa.revengine-test.org/donate/)
2. Enter a custom giving amount ie $500 for monthly contributions
3. See that you will be charged $500 if you submit your payment
4. Switch frequency to one-time or yearly frequency
5. See that the amount being charged changes to whatever is the default for the new frequency… but that frequency button is not colored in a way that shows it is being selected
6. Also see that the custom amount you entered earlier is still being shown despite it not being used

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
no

#### Have automated unit tests been added? If not, why?
yes

#### How should this change be communicated to end users?
n/a

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-2805

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
no

### CODE REVIEW

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/components/donationPage/pageContent/DAmount.test.tsx [hunk 1]
```diff
@@ -175,7 +196,8 @@ describe('DAmount', () => {
   describe('when the element has the allowOther option set', () => {
     it('displays a field where the user can enter another amount', () => {
       tree(propsWithOtherAmount);
-      expect(screen.getByRole('textbox')).toBeInTheDocument();
+      // spinbutton = <input type="number" />
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:199]: Not really, it's an explenation that role="spinbutton" relates to input with type=number. I'll make it clearer

File: spa/src/components/donationPage/pageContent/DAmount.tsx [hunk 2]
```diff
@@ -144,10 +154,12 @@ function DAmount({ element, ...props }) {
           <FeesContainer>
             <PayFeesControl
               agreedToPayFees={userAgreesToPayFees}
-              currencySymbol={page.currency.symbol}
+              currencySymbol={page.currency?.symbol ?? '$'}
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.tsx:157]: I think TS complained about this, but I'll try again.

File: spa/src/components/donationPage/pageContent/DAmount.test.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { axe } from 'jest-axe';
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:1]: You are right. I've added a test for this.

[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:1]: Sorry, I think I'm missing where you added the test?

File: spa/src/constants/contributionIntervals.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export type ContributionInterval = 'one_time' | 'month' | 'year';
 
-export const CONTRIBUTION_INTERVALS = {
+export const CONTRIBUTION_INTERVALS: { [x: string]: ContributionInterval } = {
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/constants/contributionIntervals.ts:3]: Oh, I see. I think then yeah, the type then would be something like

```ts
export type ContributionIntervalKey = 'ONE_TIME' | 'MONTHLY' | 'ANNUAL';

export const CONTRIBUTION_INTERVALS: Record<ContributionIntervalKey, ContributionInterval> = {...
```

There might be a nicer way to put it, though.

File: spa/src/components/donationPage/pageContent/DAmount.test.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:1]: To cypress. It's a test that involves multiple components

[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:1]: This file: `spa/cypress/e2e/01-donationPage.cy.js`
commit: [8ade874](https://github.com/newsrevenuehub/rev-engine/pull/937/commits/8ade87427c709db8d2e86dd4a20ac4a98b93ebfd)

[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/donationPage/pageContent/DAmount.test.tsx:1]: I see, thanks.

**Review by nrh-cklimas — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**